sscanss/app/dialogs/preferences.py

```diff
@@ -249,6 +249,36 @@ def generalForm(self):
         frame.setLayout(main_layout)
         self.stack.addWidget(create_scroll_area(frame))
 
+    def renderingSizeSlider(self, key, value, range=(5, 100)):
+        """Creates a horizontal layout containing a slider 
+            and editable value for a graphics rendering size
+        
+            :param key: a string representing a settings key
+            :type key: str
+            :param value: integer rendering size
+            :type value: int       
+            :param range: min and max values the slider can take
+            :type range: tuple
+            """
+        layout = QtWidgets.QHBoxLayout()
+
+        slider = QtWidgets.QSlider(orientation=QtCore.Qt.Orientation.Horizontal)
+        slider.setRange(range[0], range[1])
+        slider.setProperty(self.prop_name, (key, value))
+        slider.setValue(slider.property(self.prop_name)[1])
+
+        slider_value = QtWidgets.QLineEdit(str(slider.value()))
+        slider_value.setFixedWidth(40)
+        slider_value.textChanged.connect(lambda: slider.setValue(int(slider_value.text()))
```

Comment:
Please use a `QIntValidator` to enforce the textbox will only allow numbers within the given range 

![Screenshot 2023-08-01 091040](https://github.com/ISISNeutronMuon/SScanSS-2/assets/34302892/c9761c9f-e0ba-44ad-b481-3332516c390b)

Reply:
I'm a bit concerned that this constraint won't be visible for lower bound inputs since you'd need to be able to start writing an input beginning with 1 or some number below the lower bound in order to make a valid input.

I think that in this case, clicking accept with an input of 1 will simply store a lower bound.